docs/workflows.md

```diff
@@ -0,0 +1,63 @@
+
+NOTE: Regardless of the multitude of containers and mechanisms running, [the CloudAPI](http://localhost:8000) and its [SwaggerUI](http://localhost:8000/docs) are the main interaction point intended between clients and the stack. This should be the only endpoint clients should (have to) interact with. There is no need (and no intention to allow that) for client to directly interact with the webhooks or trustregistry container. For a production deployment or a close-to-proudction/smoke-testing deployment, you are well adviced to only expose this endpoint to clients and leave all other endpoints unexposed to the outside world.
+
+#### Trust registry
+
+As a client you can retrieve the trust registry (and NOT alter it). That is intential and all a consumer/client should and is able to do. Altering the trust registry is baked into admin actions and only possible with admin role.
+
+#### Webhooks
+
+A client can subscribe to the webhooks via the cloudapi (as opposed to directly via the webhooks container). This requires only targetting the `/webhooks` endpoint and optionally targetting the `/webhooks/{topic}` subroute by suffixing a topic. Using the auth mechanism (see section below), the app automatically extracts the required info about the wallet (i.e the wallet id and JWT) and retrieves only the associated webhooks with a particular wallet. Failing to authenticate will return a 403 HTTP Error.
+
+### Authentication
+
+Authentication is handled by the CloudAPI and, fom a client perspective, kept simple and convenient. Either, via the Swagger UI auth (padlock butten in UI) or via the HEADER data of your client specifying an `x-api-key`. Regardless of whether ou use the UI or another client, the `x-api-key` value consists of two parts, separated by a dot:
+
+`{role}.{key/password}`
+
+So, your header has the format `'x-api-key: {role}.{key/password}` which yields, for example, `'x-api-key: yoma.adminApiKey'`
+
+The first part `role` specifies the role on the surface and targets the correct endpoints under the hood and authentication mechanisms under the hood. The CloudAPI knows how to interpret the roles and will produce the correct target URLs for eg aca-py (`member` targets the multitenant agent) with the correct header expected by aca-py agent. For instance, `member` results in a `Bearer {TOKEN}` header against the multitenant agent whereas `member-admin` as role results in an `x-api-key` header for the multitenant agent (hence targeting the admin interface of the same multitennat agent). You may have noticed now that this mechanism also chooses which aca-py instance to target without having to know or specify the URL the agent resides under.
+
+Currently there are five options for `role`:
+
+- yoma
```

Comment:
We should explain what each role can do. E.g. yoma is the endorser, can create schemas etc.... ecosystem-admin/member-admin can only create new tenants. ecosystem can be issuer/verifier and will be automatically registered in the trust registry, while members can only be holders